hibernate-core/src/main/java/org/hibernate/cfg/EJB3DTDEntityResolver.java

```diff
@@ -114,13 +120,45 @@ else if ( systemId.endsWith( "persistence_1_0.xsd" ) ) {
 		}
 
 		// because the old code did this too (in terms of setting resolved)
-		InputSource source = super.resolveEntity( publicId, systemId );
+		InputSource source = _resolveEntity( publicId, systemId );
 		if ( source != null ) {
 			resolved = true;
 		}
 		return source;
 	}
 
+	// Copied over from DTDEntityResolver that was removed
+	private InputSource _resolveEntity(String publicId, String systemId) {
```

Comment:
Do we feel the necessity of such '_' prefixing naming convention? IMHO, it is not necessary in Java.